android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/RemoteMongoClientIntTests.kt

```diff
@@ -1080,6 +1085,294 @@ class RemoteMongoClientIntTests : BaseStitchAndroidIntTest() {
         }
     }
 
+    @Test
+    fun testWatchListenerLambda() {
+        val coll = getTestColl()
+        assertEquals(0, Tasks.await(coll.count()))
+
+        val objectId1 = ObjectId()
+        val objectId2 = ObjectId()
+
+        val rawDoc1 = Document()
+        rawDoc1["_id"] = objectId1
+        rawDoc1["hello"] = "world"
+
+        val rawDoc2 = Document()
+        rawDoc2["_id"] = objectId2
+        rawDoc2["happy"] = "day"
+
+        Tasks.await(coll.insertMany(listOf(rawDoc1, rawDoc2)))
+        assertEquals(2, Tasks.await(coll.count()))
+
+        val streamTask = coll.watch(objectId1, objectId2)
+        val stream = Tasks.await(streamTask)
+
+        var changeEventListenerSemaphore = Semaphore(0)
```

Comment:
This Semaphore + AtomicInteger mechanic can likely be replaced with a CountdownLatch